R/prevalence.R

```diff
@@ -0,0 +1,94 @@
+#' @title Taxa Prevalence 
+#' @description Simple prevalence measure.
+#' @param x A vector, data matrix or \code{\link{MicrobiomeExperiment}} object
+#' @param sort Sort the groups by prevalence
+#' @param count Logical. Indicate prevalence as fraction of samples
+#' (in percentage between 0 to 1 default); or in absolute counts indicating
+#' the number of samples where the OTU is detected (strictly) above the given
+#' abundance threshold.
+#' @inheritParams prevalentMembers
+#' @details For vectors, calculates the fraction (count=FALSE) or
+#' number (count=TRUE) of samples that exceed the
+#' detection. For matrices, calculates this for each matrix
+#' column. For MicrobiomeExperiment objects, calculates this for each OTU. The
+#' relative prevalence (count=FALSE) is simply the absolute
+#' prevalence (count=TRUE) divided by the number of samples.
+#' @return For each OTU, the fraction of samples where a given OTU is
+#' detected. The output is readily given as a percentage.
+#' @references 
+#' A Salonen et al. The adult intestinal core microbiota is determined by 
+#' analysis depth and health status. Clinical Microbiology and Infection 
+#' 18(S4):16 20, 2012. 
+#' To cite the R package, see citation('mia') 
+#' @author Leo Lahti
+#' @keywords utilities
+#' @export
+#' @examples
+#' data(enterotype)     
+#' pr <- prevalence(enterotype, detection=0, sort=TRUE, count=TRUE)
+#' pr <- prevalence(enterotype, detection=0, sort=TRUE, count=FALSE)
+prevalence <- function(x, detection=0, sort=FALSE, count=FALSE,
```

Comment:
if x is supposed to be a `SumamrizedExperiment` or derivative object, the functions, especially these important ones, must be implemented as S4 methods. If you need input on how to proceed best, please let me know.

Reply:
That's possible and I have implemented S4 functions earlier, should be fine. But how do we draw the line between important and less important ones?

Reply:
Thats usually a fine line. If it only makes sense to dispatch on a single type, it is a tied. However, to future proof it a S4 method is usually less trouble long term.

Here you have 2-3 dispatches, which you catch with an if a few lines downstream. So it makes definitely sence here.